src/Compilers/CSharp/Portable/Symbols/VarianceSafety.cs

```diff
@@ -115,7 +115,7 @@ private static void CheckPropertyVarianceSafety(PropertySymbol property, Diagnos
                 IsVarianceUnsafe(
                     property.Type,
                     requireOutputSafety: hasGetter,
-                    requireInputSafety: hasSetter,
+                    requireInputSafety: hasSetter || !(property.GetMethod?.RefKind == RefKind.None),
```

Comment:
Too bad relational rules around boo? are inconsistent with rules around bool.

---
In reply to: [101597834](https://github.com/dotnet/roslyn/pull/17179#discussion_r101597834) [](ancestors = 101597834,101596322)